ft-email-auth-provider/src/handlers/confirm_email.rs

```diff
@@ -0,0 +1,92 @@
+use crate::handlers::resend_confirmation_email::{
+    generate_new_confirmation_key, send_confirmation_email,
+};
+use validator::ValidateEmail;
+
+#[ft_sdk::form]
+pub fn confirm_email(
+    mut conn: ft_sdk::Connection,
+    ft_sdk::Query(code): ft_sdk::Query<"code">,
+    ft_sdk::Query(email): ft_sdk::Query<"email">,
+    host: ft_sdk::Host,
+    mountpoint: ft_sdk::Mountpoint,
+) -> ft_sdk::form::Result {
+    if !email.validate_email() {
+        return Err(ft_sdk::single_error("email", "invalid email format").into());
+    }
+
+    let (user_id, data) = ft_sdk::auth::provider::user_data_by_custom_attribute(
+        &mut conn,
+        crate::PROVIDER_ID,
+        crate::EMAIL_CONF_CODE_KEY,
+        &code,
+    )?;
+
+    let sent_at = data
+        .custom
+        .as_object()
+        .expect("custom is a json object")
+        .get(crate::EMAIL_CONF_SENT_AT)
+        .expect("email_conf_sent_at should exists if the account was found")
+        .as_str()
+        .expect("value must be a datetime string")
+        .parse::<chrono::DateTime<chrono::Utc>>()
+        .expect("chrono parse must work");
```

Comment:
We are using timestamp in nanoseconds which is an integer for datetime type data everywhere. So lets follow that.

Reply:
I see. I still think we should use `String`, this is not a database column but a json string in db and it'd be more readable if this is a string. Do you still want me to use nanoseconds here?

Reply:
Yes... We should follow what has been decided. Here the reason been that you can do SQL queries (like comparison operations) too if it's integer.

Reply:
https://github.com/fastn-community/auth/pull/4/commits/b162e5603d6703e233bdfa4adbfdb54ab9b278e9